run.py

```diff
@@ -0,0 +1,153 @@
+    representation_model = MistralRepresentation(nr_docs=nr_docs, diversity=document_diversity, api="generate")
+
+    return representation_model
+
+
+def main():
+    if not os.path.isfile("config.yaml"):
+        raise Exception("Missing config yaml file")
+
+    with open("config.yaml", "r") as f:
+        config = yaml.safe_load(f)
+
+    parser = argparse.ArgumentParser(description="A script with command-line arguments.")
+    parser.add_argument("-d", "--datapath", type=str, default=config.get("datapath", "./data/"), help="Path to data")
+    parser.add_argument("-e", "--embeddings", type=str, default=config.get("embeddingspath", "./data/embeddings/"), help="Path to data embeddings")
+    parser.add_argument("-r", "--result_path", type=str, default=config.get("resultspath", "./output/"), help="Path to save results")
+    # parser.add_argument("-im", "--images", action="store_true", help="Save visualizations")
+    parser.add_argument("-lm", "--low_memory", action="store_true", help="Low memory flag")
+    parser.add_argument("-n", "--nr_docs", type=int, default=config.get("nr_docs", 10),
+                        help="Number of docs per topic to pass to representation model")
+    parser.add_argument("-p", "--prompt", type=str, default=config.get("prompt", ""), help="Prompt to pass to LLM")
+    parser.add_argument("-sz", "--min_topic_size", type=int, default=config.get("min_topic_size", 100), help="Minimum topic size")
+    #
+    # # bertopic options
+    #
+    parser.add_argument("-cl", "--clustering", type=str, choices=['hbdscan', 'kmeans', 'agglomerative'],
+                        default=config.get("clustering_model", "hdbscan"), help="Clustering algorithm")
+    # parser.add_argument("-llm", "--llm", type=str, choices=['gpt4o', 'mistral-small'], help="Low memory flag")
+    parser.add_argument("-dim", "--dim_reduction", type=str, choices=['pca', 'umap'], default=config.get("dimensionality_reduction", "umap"),
+                        help="Dimensionality reduction algorithm")
+    parser.add_argument("-s", "--save_embeddings", action="store_true", default=config.get("save_embeddings", False), help="Save embeddings")
+
+    args = parser.parse_args()
+
+    print("Reading data...")
+    documents = read_data(args.datapath)
+    print("Done!")
+
+    representation_model = get_representation_model(nr_docs=args.nr_docs)
+    medical_embedding_model = get_medical_embedding_model()
+    vectorizer_model = get_count_vectorizer()
+    ctfidf_model = get_ctidf_model()
+
+    bertopic_model = BERTopicModel(ctfidf_model=ctfidf_model, embedding_model=medical_embedding_model, verbose=True,
+                                   min_topic_size=args.min_topic_size, vectorizer_model=vectorizer_model,
+                                   representation_model=representation_model, low_memory=args.low_memory)
+
+    if os.path.exists(args.embeddings):
+        print("Loading embeddings...")
```

Comment:
The if condition is repeated in the pickle command. Call `pickle.load` here if the file exists and call `medical_embedding_model` in the else clause if it doesn't.